src/tools/processData.js

```diff
@@ -265,6 +269,39 @@ export const hideRecipientsGrants = async (recipientsGrants) => {
     );
   }
   await Promise.all(promises);
+  const oldGrantNumbers = [];
+
+  for (const grant of grants) {
+    const newGrantNumber = grant.number;
+    const oldGrantNumber = await GrantNumberLink.findOne({
+      attributes: ['grantNumber'],
+      where: { grantId: grant.id, grantNumber: { [Op.ne]: grant.number } },
+    });
+    if (oldGrantNumber) {
+      oldGrantNumbers.push(oldGrantNumber.grantNumber);
+      // Update corresponding MonitoringReviewGrantee records
+      promisesMonitoring.push(
+        MonitoringReviewGrantee.update(
+          { grantNumber: newGrantNumber },
+          { where: { grantNumber: oldGrantNumber.grantNumber } },
+        ),
+      );
+      // Update corresponding MonitoringClassSummary records
+      promisesMonitoring.push(
+        MonitoringClassSummary.update(
+          { grantNumber: newGrantNumber },
+          { where: { grantNumber: oldGrantNumber.grantNumber } },
+        ),
+      );
+    }
+  }
+
+  await Promise.all(promisesMonitoring);
+
+  await GrantNumberLink.unscoped().destroy({
+    where: { grantNumber: { [Op.in]: oldGrantNumbers } },
+    force: true,
+  });
```

Comment:
GrantNumberLinks is read from, and deleted, but not repopulated. It needs to be populated.

Reply:
I think the delete should be an update

Reply:
The `GrantNumberLinks `are populated when the grants are updated. We need to delete the records with actual grant numbers once they are no longer referenced by the monitoring tables, however the obfuscated `GrantNumberLinks `will remain.